lib/config/defaults.js

```diff
@@ -262,9 +262,17 @@ const applyCacheDefaults = (cache, { name, mode }) => {
 			F(cache, "name", () => name + "-" + mode);
 			D(cache, "version", "");
 			F(cache, "cacheDirectory", () => {
-				const pkgDir = require("pkg-dir");
+				const escalade = require("escalade/sync");
 				const cwd = process.cwd();
-				const dir = pkgDir.sync(cwd);
+				// eslint-disable-next-line no-warning-comments
+				// @ts-ignore: escalade typings use es module "export default" which is treated as "{ default: escalade }" by typescript; can be fixed with migration to es modules
+				const pkgPath = escalade(cwd, (dir, names) => {
+					if (names.includes("package.json")) {
+						return "package.json";
+					}
+					return false;
+				});
+				const dir = path.dirname(pkgPath);
```

Comment:
If you want it to be even smaller, you could use this piece of code instead:

``` js
let dir = process.cwd();
for (;;) {
  try {
    if (fs.statSync(path.join(dir, "package.json").isFile()) break;
  } catch (e) {}
  const parent = path.dirname(dir);
  if (dir !== parent) { dir = undefined; break; }
}
```

Reply:
Done